onnx/helper.py

```diff
@@ -362,8 +362,7 @@ def make_tensor(
             vals = np.array(vals).astype(np_dtype).view(dtype=np.uint16).flatten().tolist()
         elif data_type == TensorProto.BFLOAT16:
             vals = list(map(float32_to_bfloat16, np.array(vals).astype(np_dtype).flatten().tolist()))
-        field = mapping.STORAGE_TENSOR_TYPE_TO_FIELD[
-            mapping.TENSOR_TYPE_TO_STORAGE_TENSOR_TYPE[data_type]]
+        field = mapping.to_field(data_type)
```

Comment:
I would instead have the function take in a tensor and a data type and returns the field. Because every single caller of this function is going to do the same `getattr(tensor, field)`, so it should just be inside the helper function.

Reply:
I introduced `get_attr_from_sequence_elem_type` and `get_attr_from_optional_elem_type` to achieve that. I will still keep `storage_type_to_field` and `optional_type_to_field` public, because I saw external code has already used them (by `STORAGE_ELEMENT_TYPE_TO_FIELD` and `OPTIONAL_ELEMENT_TYPE_TO_FIELD`). Please let me know if you still think we should hide them as private functions.